src/Gang/ui/TerritorySubpage.tsx

```diff
@@ -37,7 +38,34 @@ export function TerritorySubpage(): React.ReactElement {
           control={
             <Switch
               checked={gang.territoryWarfareEngaged}
```

Comment:
Correct me if I'm wrong, but based on what I know of React, this situation will occur:
* You try to enable warfare
* You don't have enough power, so the prompt comes up
* You select "Cancel" (don't continue with clashes)
* This component manages its own state. The *initial* checked state is based on `{gang.territoryWarfareEngaged}`, but after that it does its own thing. So now the `Switch` will be checked, but warfare is disabled.

Reply:
We are using the `Switch` component as a controlled component (https://legacy.reactjs.org/docs/forms.html#controlled-components) here. Its value is controlled by `gang.territoryWarfareEngaged`, so if we don't set that value, the `Switch` value won't be changed. For example, in the `onChange` handler, if we return immediately without changing `gang.territoryWarfareEngaged`, the `Switch` component will be stuck forever at the initial value of `gang.territoryWarfareEngaged`, even when the player tries to press it.

Video:

https://github.com/user-attachments/assets/76420df7-e365-4d6c-ab09-8ef3d2e6ae6b

Reply:
I *knew* that this was a controlled component, but I still got the semantics wrong. *Grumble grumble* stupid React *grumble grumble*.